src/cript/api/api.py

```diff
@@ -179,218 +305,195 @@ def is_vocab_valid(
+            1. Manipulates the string to be title case to work with db schema
 
         Parameters
         ----------
-        node:
-            a node in JSON form
+        node_json: str
+            a node in JSON form string
+
+        Notes
+        -----
+        This function does not take into consideration vocabulary validation.
+            For vocabulary validation please check `is_vocab_valid`
+
+        Raises
+        ------
+        CRIPTNodeSchemaError
+            in case a node is invalid
 
         Returns
         -------
         bool
             whether the node JSON is valid or not
         """
 
-        # TODO currently validate says every syntactically valid JSON is valid
-        # TODO do we want invalid schema to raise an exception?
-        node_dict = json.loads(node_json)
-        if json_validate(node_dict, self._schema):
-            return True
-        else:
-            return False
-
-    @property
-    def schema(self):
-        """
-        Access the CRIPTSchema that is associated with this API connection.
-        This can be used to validate node JSON.
-        """
-        return copy.copy(self._schema)
-
-    def connect(self):
-        """
-        Connect this API globally as the current active access point.
-        It is not necessary to call this function manually if a context manager is used.
-        A context manager is preferred where possible.
-        Jupyter notebooks are a use case where this connection can be handled manually.
-        If this function is called manually, the `API.disconnect` function has to be called later.
-
-        For manual connection: nested API object are discouraged.
-        """
-        # Store the last active global API (might be None)
-        global _global_cached_api
-        self._previous_global_cached_api = copy.copy(_global_cached_api)
-        _global_cached_api = self
-        return self
-
-    def disconnect(self):
-        """
-        Disconnect this API from the active access point.
-        It is not necessary to call this function manually if a context manager is used.
-        A context manager is preferred where possible.
-        Jupyter notebooks are a use case where this connection can be handled manually.
-        This function has to be called manually if  the `API.connect` function has to be called before.
-
-        For manual connection: nested API object are discouraged.
-        """
-        # Restore the previously active global API (might be None)
-        global _global_cached_api
-        _global_cached_api = self._previous_global_cached_api
-
-    def __enter__(self):
-        self.connect()
+        db_schema = self._get_db_schema()
 
-    def __exit__(self, type, value, traceback):
-        self.disconnect()
+        node_dict = json.loads(node_json)
+        try:
+            node_type = node_dict["node"][0].title()  # get node type from "node": ["material"]
```

Comment:
I refined this error message and catching to reflect the list expectation better.

Reply:
looks good!

after we have something working, I'm going to focus fully on UX and error messages and refine that

Reply:
I think this is resolved right?